core/core.go

```diff
@@ -382,25 +382,46 @@ func makeAddrsFactory(cfg config.Addresses) (p2pbhost.AddrsFactory, error) {
 	}, nil
 }
 
-func makeSmuxTransport(mplexExp bool) smux.Transport {
+func makeSmuxTransport(multiplexers []string, mplexExp bool) smux.Transport {
 	mstpt := mssmux.NewBlankTransport()
 
-	ymxtpt := &yamux.Transport{
-		AcceptBacklog:          512,
-		ConnectionWriteTimeout: time.Second * 10,
-		KeepAliveInterval:      time.Second * 30,
-		EnableKeepAlive:        true,
-		MaxStreamWindowSize:    uint32(1024 * 512),
-		LogOutput:              ioutil.Discard,
+	var isYamex, isMplex bool
+
+	for _, multiplexer := range multiplexers {
+		if multiplexer == "yamex" {
+			isYamex = true
+		} else if multiplexer == "mplex" {
+			isMplex = true
+		}
 	}
 
-	if os.Getenv("YAMUX_DEBUG") != "" {
-		ymxtpt.LogOutput = os.Stderr
+	if !isMplex && !isYamex {
+		isYamex = true
 	}
 
-	mstpt.AddTransport("/yamux/1.0.0", ymxtpt)
+	// If there are any multiplexers defined in the config, ignore mplexExp.
+	if len(multiplexers) > 0 {
+		mplexExp = false
+	}
+
+	if isYamex {
+		ymxtpt := &yamux.Transport{
+			AcceptBacklog:          512,
+			ConnectionWriteTimeout: time.Second * 10,
+			KeepAliveInterval:      time.Second * 30,
+			EnableKeepAlive:        true,
+			MaxStreamWindowSize:    uint32(1024 * 512),
+			LogOutput:              ioutil.Discard,
+		}
+
+		if os.Getenv("YAMUX_DEBUG") != "" {
+			ymxtpt.LogOutput = os.Stderr
+		}
+
+		mstpt.AddTransport("/yamux/1.0.0", ymxtpt)
+	}
 
-	if mplexExp {
+	if isMplex || mplexExp {
```

Comment:
It would be nice to warn if `--enable-mplex-experiment` is `false` but mplex is explicitly enabled in the config.